protected/humhub/modules/user/controllers/AuthController.php

```diff
@@ -226,7 +226,9 @@ protected function login($user, $authClient)
 
         if ($success) {
             $this->trigger(static::EVENT_AFTER_LOGIN, new UserEvent(['user' => Yii::$app->user->identity]));
-            $authClient->onSuccessLogin();
+            if (method_exists($authClient, 'onSuccessLogin')) {
```

Comment:
It would be better to use `if ($authClient instanceof \humhub\modules\user\authclient\BaseFormAuth) {` instead.

Reply:
Will work, yes, but prevents the possibility to trigger this method in a custom auth client.

Reply:
ah yes, sorry, we think you solution is better because it may works as you wrote on custom auth clients and also on old core versions where the method `onSuccessLogin()` doesn't exist yet.

Reply:
I'd also like to note that it's already possible to remove the old core OAuth clients in the future do to the new modules that were pushed to the marketplace, so in the future this might be the correct solution as of right now but I also agree that @yurabakhtin's solution is the right one to use in the future

https://www.humhub.com/en/marketplace?keyword=Sign-In